src/container/acrManager.ts

```diff
@@ -0,0 +1,237 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+"use strict";
+import { ContainerRegistryManagementClient } from "azure-arm-containerregistry";
+import { Registry, RegistryListCredentialsResult, RegistryListResult } from "azure-arm-containerregistry/lib/models";
+import { Registries } from "azure-arm-containerregistry/lib/operations";
+import * as request from "request-promise";
+import * as vscode from "vscode";
+import { Executor } from "../common/executor";
+import { UserCancelledError } from "../common/UserCancelledError";
+import { AcrRegistryQuickPickItem } from "../container/models/AcrRegistryQuickPickItem";
+import { AzureAccount, AzureSession, AzureSubscription } from "../typings/azure-account.api";
+
+export class AcrManager {
+    private readonly azureAccount: AzureAccount;
+    private acrRefreshToken: string;
+
+    constructor() {
+        this.azureAccount = vscode.extensions.getExtension<AzureAccount>("ms-vscode.azure-account")!.exports;
+    }
+
+    public async selectAcrImage(): Promise<string> {
+        const acrRegistryItem: AcrRegistryQuickPickItem = await this.selectAcrRegistry();
```

Comment:
suggest to get the ContainerRegistryManagementClient first in this method. Thus the client could be reused in the flow of this method.